inc/namespace.php

```diff
@@ -1,9 +1,115 @@
 <?php
+/**
+ * Core platform functions.
+ *
+ * @package hm-platform
+ */
 
 namespace HM\Platform;
 
 use Aws\Sdk;
 
+/**
+ * Retrieve the configuration for HM Platform.
+ *
+ * The configuration is defined by merging the defaults set by modules
+ * with any overrides present in composer.json.
+ *
+ * @return array Configuration data.
+ */
+function get_config() : array {
+	static $config = [];
+
+	if ( empty( $config ) ) {
+		$config = get_merged_config();
+	}
+
+	/**
+	 * Filter the entire platform config.
+	 *
+	 * @param array $config The full config array.
+	 */
+	$config = apply_filters( 'hm-platform.config', $config );
+
+	return $config;
+}
+
+/**
+ * Merge the defaults and the contents of the various configuration files into a single configuration.
+ *
+ * @return array Configuration data.
+ */
+function get_merged_config() : array {
+	// Path to composer file.
+	$composer_file = ROOT_DIR . '/composer.json';
+	$composer_json = get_json_file_contents_as_array( $composer_file );
+	$config = merge_config_settings( [], $composer_json['extra']['platform'] ?? [] );
+
+	// Look for environment specific settings in the config and merge it in.
+	$environment = get_environment_type();
+	$config = merge_config_settings( $config, $config['environments'][ $environment ] ?? [] );
+
+	return $config;
+}
+
+/**
+ * Merge settings in an existing configuration file.
+ *
+ * @param array $config    Existing configuration.
+ * @param array $overrides Settings to merge in.
+ *
+ * @return array Configuration data.
+ */
+function merge_config_settings( array $config, array $overrides ) : array {
+	foreach ( $overrides as $key => $value ) {
+		switch ( $key ) {
+			// Merge module settings together.
+			case 'modules':
+				foreach ( $value as $module => $settings ) {
+					$config[ $key ][ $module ] = array_merge( $config[ $key ][ $module ] ?? [], $settings );
+				}
+				break;
+			// Replace property by default.
+			default:
+				$config[ $key ] = $value;
+				break;
+		}
+	}
+
+	return $config;
+}
+
+/**
+ * Get the contents of a JSON file, decode it, and return as an array.
+ *
+ * @param string $file Path to the JSON file.
+ *
+ * @return array Decoded data in array form, empty array if JSON data could not read.
+ */
+function get_json_file_contents_as_array( $file ) : array {
+	if ( ! strpos( $file, '.json' ) ) {
+		// phpcs:ignore
+		trigger_error( $file . ' is not a JSON file.', E_USER_WARNING );
+		return [];
+	}
+
+	if ( ! is_readable( $file ) ) {
+		// phpcs:ignore
+		trigger_error( 'Could not read ' . $file . ' file.', E_USER_WARNING );
+		return [];
+	}
+
+	$contents = json_decode( file_get_contents( $file ), true );
+
+	if ( ! is_array( $contents ) ) {
```

Comment:
Should use `json_last_error() !== JSON_ERROR_NONE` instead, and maybe `json_last_error_msg()` for the message.